soc/arm/microchip_mec/mec1501/device_power.c

```diff
@@ -194,23 +191,128 @@ static void deep_sleep_restore_timers(void)
 
 	p = &ds_timer_tbl[0];
 	for (i = 0; i < NUM_DS_TIMER_ENTRIES; i++) {
-		REG32(p->addr) = timers[i] & ~p->restore_mask;
+		REG32(p->addr) = ds_ctx.timers[i] & ~p->restore_mask;
 		p++;
 	}
 }
 
+#ifdef DEEP_SLEEP_PERIPH_SAVE_RESTORE_EXTENDED
+
+static void deep_sleep_save_blocks(void)
+{
+	/* ADC Power saving feature is enabled */
+	ds_ctx.adc_info.adc_ctrl = ADC_REGS->CONTROL;
+	ds_ctx.adc_info.adc_cfg = ADC_REGS->CONFIG;
+	ADC_REGS->CONTROL = 0;
+	ADC_REGS->CONFIG = 0;
+
+	ds_ctx.peci_info.peci_ctrl = PECI_REGS->CONTROL;
+	ds_ctx.peci_info.peci_dis = ECS_REGS->PECI_DIS;
+	PECI_REGS->CONTROL = 0x08;
+	PECI_REGS->CONTROL = 0x09;
+	ECS_REGS->PECI_DIS = 0x01;
+
+	/* SMBUS */
+	ds_ctx.smb_info[0] = SMB0_REGS->CTRLSTS;
+	ds_ctx.smb_info[1] = SMB1_REGS->CTRLSTS;
+	ds_ctx.smb_info[2] = SMB2_REGS->CTRLSTS;
+	ds_ctx.smb_info[3] = SMB3_REGS->CTRLSTS;
+	ds_ctx.smb_info[4] = SMB4_REGS->CTRLSTS;
+
+	mchp_pcr_periph_slp_ctrl(PCR_SMB1, MCHP_PCR_SLEEP_EN);
+	mchp_pcr_periph_slp_ctrl(PCR_SMB2, MCHP_PCR_SLEEP_EN);
+	mchp_pcr_periph_slp_ctrl(PCR_SMB2, MCHP_PCR_SLEEP_EN);
+	mchp_pcr_periph_slp_ctrl(PCR_SMB3, MCHP_PCR_SLEEP_EN);
+	mchp_pcr_periph_slp_ctrl(PCR_SMB4, MCHP_PCR_SLEEP_EN);
+
+	mchp_pcr_periph_slp_ctrl(PCR_ESPI, MCHP_PCR_SLEEP_EN);
+
+	/* Comparators */
+	ds_ctx.comp_info = ECS_REGS->CMP_SLP_CTRL;
+	ECS_REGS->CMP_SLP_CTRL = 0;
+
+	/* Disable VCI_INPUT_ENABLE */
+	ds_ctx.vci_info = VCI_REGS->INPUT_EN;
+	VCI_REGS->INPUT_EN = 0;
+
+	/* Disable WEEK TIMER */
+	ds_ctx.wktmr_info = WKTMR_REGS->BGPO_PWR;
```

Comment:
This is not disabling the week timer.  It's setting the BGPO pins to GPIO mode where they are controlled by their respective GPIO CONTROL registers.

Reply:
@scottwcpg ok, will remove this since this is already done in board pinmux.c